### ISSUE TITLE
Missed SLP vectorization with AND op with constant

### ISSUE THREAD

**Issue by davidbolvansky** [llvm/llvm-project#49278]
|  |  |
| --- | --- |
| Bugzilla Link | [49934](https://llvm.org/bz49934) |
| Version | trunk |
| OS | Windows NT |
| CC | @topperc,@fhahn,@RKSimon,@rotateright |

## Extended Description 
```
void baz( uint8_t *__restrict dst, uint8_t *__restrict src) {    
	for( int x = 0; x < 8; x++ ){
        dst[x] = src[x]&(~63);
    }
}
```
LLVM -O3
```
baz(unsigned char*, unsigned char*):                             # @baz(unsigned char*, unsigned char*)
        mov     al, byte ptr [rsi]
        and     al, -64
        mov     byte ptr [rdi], al
        mov     al, byte ptr [rsi + 1]
        and     al, -64
        mov     byte ptr [rdi + 1], al
        mov     al, byte ptr [rsi + 2]
        and     al, -64
        mov     byte ptr [rdi + 2], al
        mov     al, byte ptr [rsi + 3]
        and     al, -64
        mov     byte ptr [rdi + 3], al
        mov     al, byte ptr [rsi + 4]
        and     al, -64
        mov     byte ptr [rdi + 4], al
        mov     al, byte ptr [rsi + 5]
        and     al, -64
        mov     byte ptr [rdi + 5], al
        mov     al, byte ptr [rsi + 6]
        and     al, -64
        mov     byte ptr [rdi + 6], al
        mov     al, byte ptr [rsi + 7]
        and     al, -64
        mov     byte ptr [rdi + 7], al
        ret
```
ICC/GCC:
```
baz(unsigned char*, unsigned char*):
        vmovq   xmm0, QWORD PTR [rsi]
        vmovq   xmm1, QWORD PTR .LC0[rip]
        vpand   xmm0, xmm0, xmm1
        vmovq   QWORD PTR [rdi], xmm0
        ret
```
https://godbolt.org/z/on8TanTa8

**Comment by RKSimon** [comment]
Another sub-128-bit vector case -  D124284 should handle this

**Comment by davidbolvansky** [comment]
Thanks for commiting tests

**Comment by RKSimon** [comment]
Fixed in 9dc4ced204d1e918d4c8d3279e52197e9a5abc94
